wallet-new/src/Cardano/Wallet/Kernel/Transactions.hs

```diff
@@ -229,7 +229,12 @@ newUnsignedTransaction ActiveWallet{..} options accountId payees = runExceptT $
     -- that it may change in the future.
     let attributes = def :: TxAttributes
     let tx = UnsignedTx inputs outputs attributes coins
-    return (snapshot, tx, availableUtxo)
+
+    -- STEP 3: sanity test.
+    let fees = computeFeesOfUnsignedTx tx
+    if isSane options fees
+    then return (snapshot, tx, fees, availableUtxo)
+    else throwError $ NewTransactionErrorCoinSelectionFailed CoinSelHardErrCannotCoverFee
```

Comment:
The error would be a bit misleading here and I wouldn't define any particular new user-facing error either; this is an invariant violation. If we reach that case, it means we fucked up hard and everything should blow up!